src/runtime/hexagon_cache_allocator.cpp

```diff
@@ -0,0 +1,165 @@
+#include "HalideRuntime.h"
+#include "mini_hexagon_dma.h"
+#include "HalideRuntimeHexagonDma.h"
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+typedef struct hexagon_cache_block {
+    void *l2memory;
+    size_t bytes;
+    bool used;
+    struct hexagon_cache_block *next;
+} hexagon_cache_pool_t;
+
+typedef hexagon_cache_pool_t *pcache_pool;
+
+pcache_pool hexagon_cache_pool = NULL;
+halide_mutex hexagon_cache_mutex; 
+
+}}}
+
+using namespace Halide::Runtime::Internal;
+
+extern "C" {
+//TO DO Add hjalide_mutex_lock to make it thread sahe 
+static inline void *hexagon_cache_pool_get (void *user_context, size_t size, bool retry) {
```

Comment:
Add comment explaining retry logic.

Reply:
Added a comment to explain retry logic. It is currently set to false by default as we are worried it may deallocate blocks still needed. Will need to look at when we can enable this logic.